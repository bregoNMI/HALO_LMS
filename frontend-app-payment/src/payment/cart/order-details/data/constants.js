// Product types
export const COURSE_ENTITLEMENT_PRODUCT_TYPE = 'Course Entitlement';
export const ENROLLMENT_CODE_PRODUCT_TYPE = 'Enrollment Code';
export const SEAT_PRODUCT_TYPE = 'Seat';

// Certificate types
export const VERIFIED_CERTIFICATE_TYPE = 'verified';
export const CREDIT_CERTIFICATE_TYPE = 'credit';

// Order Details message keys
export const ENTITLEMENT_MESSAGE_KEY = 'entitlement';
export const ENROLLMENT_CODE_MESSAGE_KEY = 'enrollment.code';
export const VERIFIED_SEAT_MESSAGE_KEY = 'seat.verified';
export const CREDIT_SEAT_MESSAGE_KEY = 'seat.credit';
export const SEAT_MESSAGE_KEY = 'seat';
