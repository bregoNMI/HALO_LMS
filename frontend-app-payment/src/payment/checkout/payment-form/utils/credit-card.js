import {
  faCcAmex,
  faCcDinersClub,
  faCcDiscover,
  faCcJcb,
  faCcMastercard,
  faCcVisa,
} from '@fortawesome/free-brands-svg-icons';

const CardValidator = require('../../card-validator');

export const CARD_TYPES = {
  'american-express': '003',
  'diners-club': '005',
  discover: '004',
  jcb: '007',
  mastercard: '002',
  visa: '001',
};

export const CARD_ICONS = {
  '003': faCcAmex,
  '005': faCcDinersClub,
  '004': faCcDiscover,
  '007': faCcJcb,
  '002': faCcMastercard,
  '001': faCcVisa,
};

export function getCardIconForType(cardType) {
  let cardIcon = null;
  if (CARD_ICONS[cardType] !== undefined) {
    cardIcon = CARD_ICONS[cardType];
  }
  return cardIcon;
}

export function getCardTypeId(cardNumber) {
  const { card } = CardValidator.number(cardNumber);
  if (card && CARD_TYPES[card.type] !== undefined) {
    return CARD_TYPES[card.type];
  }
  return null;
}

export function getCardIcon(cardNumber) {
  const cardType = getCardTypeId(cardNumber);
  if (cardType === null) {
    return null;
  }
  return getCardIconForType(cardType);
}
