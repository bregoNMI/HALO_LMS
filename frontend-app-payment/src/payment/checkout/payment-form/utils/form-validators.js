// eslint-disable-next-line import/no-extraneous-dependencies
import { State } from 'country-state-city';

export const getCountryStatesMap = (country) => {
  const states = State.getStatesOfCountry(country);

  if (!states.length) {
    return null;
  }
  const statesMap = {};
  states.forEach((state) => {
    statesMap[state.isoCode] = state.name;
  });
  return country && statesMap;
};

// eslint-disable-next-line import/prefer-default-export
export function isPostalCodeRequired(selectedCountry) {
  const countryListRequiredPostalCode = ['CA', 'GB', 'US'];
  const postalCodeRequired = countryListRequiredPostalCode.includes(selectedCountry);

  return postalCodeRequired;
}

export function getRequiredFields(fieldValues, isBulkOrder = false, enableStripePaymentProcessor = false) {
  const {
    firstName,
    lastName,
    address,
    city,
    country,
    state,
    postalCode,
    cardExpirationMonth,
    cardExpirationYear,
    organization,
  } = fieldValues;

  const requiredFields = {
    firstName,
    lastName,
    address,
    city,
    country,
  };

  if (!enableStripePaymentProcessor) {
    requiredFields.cardExpirationMonth = cardExpirationMonth;
    requiredFields.cardExpirationYear = cardExpirationYear;
  }

  if (isPostalCodeRequired(country) && enableStripePaymentProcessor) {
    requiredFields.postalCode = postalCode;
  }

  // By using the country-state-city library to populate states, every country that
  //  has states from the ISO 3166-2 list will have states as a required field
  if (getCountryStatesMap(country)) {
    requiredFields.state = state;
  }

  if (isBulkOrder) {
    requiredFields.organization = organization;
  }

  return requiredFields;
}

export function validateAsciiNames(firstName, lastName) {
  const errors = {};

  if (
    firstName
      && lastName
      && !/[A-Za-z]/.test(firstName + lastName)
  ) {
    errors.firstName = 'payment.form.errors.ascii.name';
  }

  return errors;
}

export function validateRequiredFields(values) {
  const errors = {};

  Object.keys(values).forEach((key) => {
    if (!values[key]) {
      errors[key] = 'payment.form.errors.required.field';
    }
  });

  return errors;
}

export function validateCardDetails(cardExpirationMonth, cardExpirationYear) {
  const errors = {};

  const currentMonth = new Date().getMonth() + 1;
  const currentYear = new Date().getFullYear();
  if (
    cardExpirationMonth
      && parseInt(cardExpirationMonth, 10) < currentMonth
      && parseInt(cardExpirationYear, 10) === currentYear
  ) {
    errors.cardExpirationMonth = 'payment.form.errors.card.expired';
  }

  return errors;
}
