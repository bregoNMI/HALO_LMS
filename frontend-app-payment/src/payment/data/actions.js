import { createRoutine } from 'redux-saga-routines';

// Routines are action + action creator pairs in a series.
// Actions adhere to the flux standard action format.
// Routines by default are in the form of:
//
// Action                |   Action Creator
// -----------------------------------------------
// fetchBasket.TRIGGER   |   fetchBasket()
// fetchBasket.SUCCESS   |   fetchBasket.success()
// fetchBasket.FAILURE   |   fetchBasket.failure()
// fetchBasket.FULFILL   |   fetchBasket.fulfill()
//
// Created with redux-saga-routines
export const fetchCaptureKey = createRoutine('FETCH_CAPTURE_KEY');
export const fetchClientSecret = createRoutine('FETCH_CLIENT_SECRET');
export const submitPayment = createRoutine('SUBMIT_PAYMENT');
export const fetchBasket = createRoutine('FETCH_BASKET');
export const addCoupon = createRoutine('ADD_COUPON');
export const removeCoupon = createRoutine('REMOVE_COUPON');
export const updateQuantity = createRoutine('UPDATE_QUANTITY');

// Actions and their action creators
export const BASKET_DATA_RECEIVED = 'BASKET_DATA_RECEIVED';

export const basketDataReceived = basket => ({
  type: BASKET_DATA_RECEIVED,
  payload: basket,
});

export const BASKET_PROCESSING = 'BASKET_PROCESSING';

export const basketProcessing = isProcessing => ({
  type: BASKET_PROCESSING,
  payload: isProcessing,
});

export const CAPTURE_KEY_PROCESSING = 'CAPTURE_KEY_PROCESSING';

export const captureKeyProcessing = isProcessing => ({
  type: CAPTURE_KEY_PROCESSING,
  payload: isProcessing,
});

export const CLIENT_SECRET_PROCESSING = 'CLIENT_SECRET_PROCESSING';

export const clientSecretProcessing = isProcessing => ({
  type: CLIENT_SECRET_PROCESSING,
  payload: isProcessing,
});

export const MICROFORM_STATUS = 'MICROFORM_STATUS';

export const microformStatus = status => ({
  type: MICROFORM_STATUS,
  payload: status,
});

export const CAPTURE_KEY_START_TIMEOUT = 'CAPTURE_KEY_START_TIMEOUT';

export const captureKeyStartTimeout = () => ({
  type: CAPTURE_KEY_START_TIMEOUT,
});
export const CAPTURE_KEY_DATA_RECEIVED = 'CAPTURE_KEY_DATA_RECEIVED';

export const captureKeyDataReceived = captureKey => ({
  type: CAPTURE_KEY_DATA_RECEIVED,
  payload: captureKey,
});

export const CLIENT_SECRET_DATA_RECEIVED = 'CLIENT_SECRET_DATA_RECEIVED';

export const clientSecretDataReceived = clientSecret => ({
  type: CLIENT_SECRET_DATA_RECEIVED,
  payload: clientSecret,
});
