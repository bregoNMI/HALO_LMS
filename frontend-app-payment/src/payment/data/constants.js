export const ORDER_TYPES = {
  BULK_ENROLLMENT: 'Enrollment Code',
  ENTITLEMENT: 'Course Entitlement',
  SEAT: 'Seat',
};

export const CERTIFICATE_TYPES = {
  VERIFIED: 'verified',
  CREDIT: 'credit',
};
