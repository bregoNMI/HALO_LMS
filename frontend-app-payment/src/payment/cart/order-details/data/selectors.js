import { createSelector, createStructuredSelector } from 'reselect';

import { getModuleState } from '../../../data/utils';
import {
  SEAT_MESSAGE_KEY,
  CREDIT_SEAT_MESSAGE_KEY,
  VERIFIED_SEAT_MESSAGE_KEY,
  SEAT_PRODUCT_TYPE,
  VERIFIED_CERTIFICATE_TYPE,
  CREDIT_CERTIFICATE_TYPE,
  ENROLLMENT_CODE_PRODUCT_TYPE,
  COURSE_ENTITLEMENT_PRODUCT_TYPE,
  ENTITLEMENT_MESSAGE_KEY,
  ENROLLMENT_CODE_MESSAGE_KEY,
} from './constants';

export const storePath = ['payment', 'basket'];

const paymentSelector = (state, props) => getModuleState(state, props.storePath ? props.storePath : storePath);

const lastProductSelector = createSelector(
  paymentSelector,
  ({ products }) => products[products.length - 1],
);

const messageTypeSelector = createSelector(
  lastProductSelector,
  (lastProduct) => {
    if (lastProduct === undefined) {
      return null;
    }
    const { productType, certificateType } = lastProduct;
    if (productType === COURSE_ENTITLEMENT_PRODUCT_TYPE) {
      return ENTITLEMENT_MESSAGE_KEY;
    }
    if (productType === ENROLLMENT_CODE_PRODUCT_TYPE) {
      return ENROLLMENT_CODE_MESSAGE_KEY;
    }
    if (productType === SEAT_PRODUCT_TYPE) {
      if (certificateType === VERIFIED_CERTIFICATE_TYPE) {
        return VERIFIED_SEAT_MESSAGE_KEY;
      }
      if (certificateType === CREDIT_CERTIFICATE_TYPE) {
        return CREDIT_SEAT_MESSAGE_KEY;
      }
      return SEAT_MESSAGE_KEY;
    }
    return null;
  },
);

export const orderDetailsMapStateToProps = createStructuredSelector({
  messageType: messageTypeSelector,
});
