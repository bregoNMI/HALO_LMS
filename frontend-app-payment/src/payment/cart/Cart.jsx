import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { FormattedMessage, injectIntl, intlShape } from '@edx/frontend-platform/i18n';
import { Collapsible } from '@openedx/paragon';

import messages from './Cart.messages';
import { cartSelector, currencyDisclaimerSelector } from '../data/selectors';
import { ORDER_TYPES } from '../data/constants';

import BulkOrderSummaryTable from './BulkOrderSummaryTable';
import CartSkeleton from './CartSkeleton';
import CartContents from './CartContents';
import CouponForm from './CouponForm';
import { CurrencyDisclaimer } from './CurrencyDisclaimer';
import OrderSummary from './OrderSummary';
import OrderDetails from './order-details';
import Offers from './Offers';
import ProductLineItem from './ProductLineItem';
import SummaryTable from './SummaryTable';
import TotalTable from './TotalTable';
import UpdateQuantityForm from './UpdateQuantityForm';

class Cart extends React.Component {
  renderCart() {
    const {
      products,
      orderType,
      isCurrencyConverted,
      isNumEnrolledExperiment,
      REV1045Experiment,
      isTransparentPricingExperiment,
      enrollmentCountData,
      orderTotal,
      showCouponForm,
      summaryPrice,
      summarySubtotal,
      summaryQuantity,
      summaryDiscounts,
      offers,
      loaded,
    } = this.props;

    const isBulkOrder = orderType === ORDER_TYPES.BULK_ENROLLMENT;

    return (
      <div>
        <span className="sr-only">
          <FormattedMessage
            id="payment.screen.reader.cart.details.loaded"
            defaultMessage="Shopping cart details are loaded."
            description="Screen reader text to be read when cart details load."
          />
        </span>

        <CartContents>
          {products.map(product => (
            <ProductLineItem
              key={product.title}
              isNumEnrolledExperiment={isNumEnrolledExperiment}
              REV1045Experiment={REV1045Experiment}
              enrollmentCountData={enrollmentCountData}
              {...product}
            />
          ))}

          {isBulkOrder ? <UpdateQuantityForm /> : null}
        </CartContents>

        {!REV1045Experiment || loaded
          ? (
            <OrderSummary> {isBulkOrder ? (
              <BulkOrderSummaryTable
                price={summaryPrice}
                subtotal={summarySubtotal}
                quantity={summaryQuantity}
              />
            ) : (<SummaryTable price={summaryPrice} />)}
              <Offers
                discounts={summaryDiscounts}
                offers={offers}
                isBundle={products.length > 1}
              />

              {showCouponForm
                ? <CouponForm /> : null}

              <TotalTable total={orderTotal} />

              {isCurrencyConverted ? (
                <CurrencyDisclaimer
                  currencyDisclaimerSelector={currencyDisclaimerSelector}
                />
              ) : null}
            </OrderSummary>
          ) : (
            <>
              <div className="skeleton py-2 mb-3 w-50" />
              <div className="skeleton py-2 mb-2" />
              <div className="skeleton py-2 mb-5" />
            </>
          )}

        {isTransparentPricingExperiment
          ? (
            <Collapsible
              styling="basic"
              title="Fair-Price Promise"
              className="pb-5 pt-1 mt-n5"
              id="fair-price-collapsible"
            >
              <p style={{ textAlign: 'justify' }}>Major brands in online education markup their products 2-5x the actual cost due to subscription pricing. We do things differently. To make online education accessible to everyone everywhere, we keep our pricing simple and transparent. A one-time investment will bring you an array of possibilities; whether entering the job market, changing fields, seeking promotion or exploring new interests, edX delivers courses for curious minds with an affordable price tag.</p>
            </Collapsible>
          ) : null }

        <OrderDetails REV1045Experiment={REV1045Experiment} products={products} />
      </div>
    );
  }

  render() {
    const {
      intl,
      loading,
      REV1045Experiment,
    } = this.props;

    return (
      <section
        aria-live="polite"
        aria-relevant="all"
        aria-label={intl.formatMessage(messages['payment.section.cart.label'])}
      >
        {!REV1045Experiment && loading ? <CartSkeleton /> : this.renderCart() }
      </section>
    );
  }
}

Cart.propTypes = {
  intl: intlShape.isRequired,
  isNumEnrolledExperiment: PropTypes.bool,
  REV1045Experiment: PropTypes.bool,
  isTransparentPricingExperiment: PropTypes.bool,
  enrollmentCountData: PropTypes.arrayOf(PropTypes.shape({
    key: PropTypes.string,
    enrollment_count: PropTypes.number,
  })),
  loading: PropTypes.bool,
  loaded: PropTypes.bool,
  products: PropTypes.arrayOf(PropTypes.shape({
    imageUrl: PropTypes.string,
    title: PropTypes.string,
    certificateType: PropTypes.oneOf(['audit', 'honor', 'verified', 'no-id-professional', 'professional', 'credit']),
  })),
  showCouponForm: PropTypes.bool,
  orderType: PropTypes.oneOf(Object.values(ORDER_TYPES)),
  isCurrencyConverted: PropTypes.bool,
  orderTotal: PropTypes.number,
  offers: PropTypes.arrayOf(PropTypes.shape({
    benefitType: PropTypes.oneOf(['Percentage', 'Absolute']).isRequired,
    benefitValue: PropTypes.number.isRequired,
    provider: PropTypes.string,
  })),
  summarySubtotal: PropTypes.number,
  summaryQuantity: PropTypes.number,
  summaryDiscounts: PropTypes.number,
  summaryPrice: PropTypes.number,
};

Cart.defaultProps = {
  isNumEnrolledExperiment: false,
  REV1045Experiment: false,
  isTransparentPricingExperiment: false,
  enrollmentCountData: null,
  loading: true,
  loaded: false,
  products: [],
  orderType: ORDER_TYPES.SEAT,
  showCouponForm: false,
  isCurrencyConverted: false,
  orderTotal: undefined,
  offers: [],
  summarySubtotal: undefined,
  summaryQuantity: undefined,
  summaryDiscounts: undefined,
  summaryPrice: undefined,
};

export default connect(cartSelector)(injectIntl(Cart));
