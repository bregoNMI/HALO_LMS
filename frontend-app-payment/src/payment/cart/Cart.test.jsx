/* eslint-disable react/jsx-no-constructed-context-values */
import React from 'react';
import { render } from '@testing-library/react';
import { act } from 'react-dom/test-utils';
import { Provider } from 'react-redux';
import { IntlProvider } from '@edx/frontend-platform/i18n';
import { AppContext } from '@edx/frontend-platform/react';
import { Factory } from 'rosie';
import { createStore } from 'redux';
import Cookies from 'universal-cookie';

import '../__factories__/basket.factory';
import '../__factories__/userAccount.factory';
import Cart from './Cart';
import createRootReducer from '../../data/reducers';
import { fetchBasket, basketDataReceived } from '../data/actions';
import { transformResults } from '../data/utils';

jest.mock('@edx/frontend-platform/analytics', () => ({
  sendTrackEvent: jest.fn(),
}));
jest.mock('universal-cookie', () => {
  class MockCookies {
    static result = {};

    get() {
      return MockCookies.result;
    }
  }
  return MockCookies;
});

describe('<Cart />', () => {
  let store;
  let tree;

  beforeEach(() => {
    const authenticatedUser = Factory.build('userAccount');
    Cookies.result = undefined;
    store = createStore(createRootReducer(), {});

    const component = (
      <IntlProvider locale="en">
        <AppContext.Provider value={{ authenticatedUser }}>
          <Provider store={store}>
            <Cart />
          </Provider>
        </AppContext.Provider>
      </IntlProvider>
    );
    tree = render(component);
  });

  it('renders the loading skeleton', () => {
    expect(tree.container).toMatchSnapshot();
  });

  it('renders a basic, one product cart', () => {
    act(() => {
      store.dispatch(basketDataReceived(transformResults(Factory.build(
        'basket',
        {},
        { numProducts: 1 },
      ))));
      store.dispatch(fetchBasket.fulfill());
    });

    expect(tree.container).toMatchSnapshot();
  });

  it('renders a basic, one product cart with coupon form', () => {
    act(() => {
      store.dispatch(basketDataReceived(transformResults(Factory.build(
        'basket',
        { show_coupon_form: true },
        { numProducts: 1 },
      ))));
      store.dispatch(fetchBasket.fulfill());
    });

    expect(tree.container).toMatchSnapshot();
  });

  it('renders a cart with an offer', () => {
    act(() => {
      store.dispatch(basketDataReceived(transformResults(Factory.build(
        'basket',
        {
          offers: [{ benefitValue: 50, benefitType: 'Percentage', provider: 'Pied Piper' }],
        },
        { numProducts: 1 },
      ))));
      store.dispatch(fetchBasket.fulfill());
    });

    expect(tree.container).toMatchSnapshot();
  });

  it('renders a cart in non USD currency', () => {
    Cookies.result = {
      code: 'MXN',
      rate: 19.092733,
    };
    // This test uses its own setup since we don't have actions to update currency.
    store = createStore(
      createRootReducer(),
      {
        payment: {},
      },
    );
    const component = (
      <IntlProvider locale="en">
        <Provider store={store}>
          <Cart />
        </Provider>
      </IntlProvider>
    );
    tree = render(component);

    act(() => {
      store.dispatch(basketDataReceived(transformResults(Factory.build(
        'basket',
        {},
        { numProducts: 1 },
      ))));
      store.dispatch(fetchBasket.fulfill());
    });

    expect(tree.container).toMatchSnapshot();
  });

  it('renders a bulk enrollment cart', () => {
    act(() => {
      store.dispatch(basketDataReceived(transformResults(Factory.build(
        'basket',
        {},
        { numProducts: 1, productType: 'Enrollment Code' },
      ))));
      store.dispatch(fetchBasket.fulfill());
    });

    expect(tree.container).toMatchSnapshot();
  });
});
