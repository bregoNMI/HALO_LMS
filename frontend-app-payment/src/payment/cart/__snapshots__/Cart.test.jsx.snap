// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cart /> renders a basic, one product cart 1`] = `
<div>
  <section
    aria-label="Shopping Cart Details"
    aria-live="polite"
    aria-relevant="all"
  >
    <div>
      <span
        class="sr-only"
      >
        Shopping cart details are loaded.
      </span>
      <div
        class="basket-section"
      >
        <h5
          aria-level="2"
        >
          In Your Cart
        </h5>
        <p>
          Your purchase contains the following:
        </p>
        <div
          class="row align-items-center mb-3 product-wrapper"
        >
          <div
            class="col-5"
          >
            <div
              class="embed-responsive embed-responsive-16by9"
            >
              <img
                alt=""
                class="img-thumbnail product-thumbnail"
                src="https://edx.org/vot.png"
              />
            </div>
          </div>
          <div
            class="col-7"
          >
            <h6
              aria-level="3"
              class="m-0"
            >
              Introduction to Llamass
            </h6>
            <p
              class="m-0"
            >
              Verified Certificate
            </p>
          </div>
        </div>
      </div>
      <div
        aria-labelledby="summary-heading"
        aria-live="polite"
        class="basket-section"
        role="region"
      >
        <h5
          aria-level="2"
          id="summary-heading"
        >
          Summary
        </h5>
         
        <div
          class="summary-row d-flex"
        >
          <span
            class="flex-grow-1"
          >
            Price
          </span>
          <span
            class="summary-price"
          >
            $464.88
          </span>
        </div>
        <div
          class="summary-row font-weight-bold d-flex"
        >
          <span
            class="flex-grow-1"
          >
            TOTAL
          </span>
          <span
            class="text-right"
          >
            $464.88
          </span>
        </div>
      </div>
      <div
        class="basket-section"
      >
        <h5
          aria-level="2"
        >
          Order Details
        </h5>
        <p>
          After you complete your order you will be able to select course dates from your dashboard.
        </p>
      </div>
    </div>
  </section>
</div>
`;

exports[`<Cart /> renders a basic, one product cart with coupon form 1`] = `
<div>
  <section
    aria-label="Shopping Cart Details"
    aria-live="polite"
    aria-relevant="all"
  >
    <div>
      <span
        class="sr-only"
      >
        Shopping cart details are loaded.
      </span>
      <div
        class="basket-section"
      >
        <h5
          aria-level="2"
        >
          In Your Cart
        </h5>
        <p>
          Your purchase contains the following:
        </p>
        <div
          class="row align-items-center mb-3 product-wrapper"
        >
          <div
            class="col-5"
          >
            <div
              class="embed-responsive embed-responsive-16by9"
            >
              <img
                alt=""
                class="img-thumbnail product-thumbnail"
                src="https://edx.org/watpewal.gif"
              />
            </div>
          </div>
          <div
            class="col-7"
          >
            <h6
              aria-level="3"
              class="m-0"
            >
              Introduction to Pigeonss
            </h6>
            <p
              class="m-0"
            >
              Verified Certificate
            </p>
          </div>
        </div>
      </div>
      <div
        aria-labelledby="summary-heading"
        aria-live="polite"
        class="basket-section"
        role="region"
      >
        <h5
          aria-level="2"
          id="summary-heading"
        >
          Summary
        </h5>
         
        <div
          class="summary-row d-flex"
        >
          <span
            class="flex-grow-1"
          >
            Price
          </span>
          <span
            class="summary-price"
          >
            $445.19
          </span>
        </div>
        <form
          class="summary-row d-flex align-items-end"
        >
          <div
            class="pgn__form-group mb-0 mr-2"
          >
            <label
              class="h6 d-block"
              for="couponField"
            >
              Add coupon code (optional)
            </label>
            <div
              class="pgn__form-control-decorator-group"
            >
              <input
                class="form-control"
                data-testid="couponField"
                id="couponField"
                name="couponField"
                type="text"
                value=""
              />
            </div>
          </div>
          <button
            class="btn btn-outline-primary"
            type="submit"
          >
            Apply
          </button>
        </form>
        <div
          class="summary-row font-weight-bold d-flex"
        >
          <span
            class="flex-grow-1"
          >
            TOTAL
          </span>
          <span
            class="text-right"
          >
            $445.19
          </span>
        </div>
      </div>
      <div
        class="basket-section"
      >
        <h5
          aria-level="2"
        >
          Order Details
        </h5>
        <p>
          After you complete your order you will be able to select course dates from your dashboard.
        </p>
      </div>
    </div>
  </section>
</div>
`;

exports[`<Cart /> renders a bulk enrollment cart 1`] = `
<div>
  <section
    aria-label="Shopping Cart Details"
    aria-live="polite"
    aria-relevant="all"
  >
    <div>
      <span
        class="sr-only"
      >
        Shopping cart details are loaded.
      </span>
      <div
        class="basket-section"
      >
        <h5
          aria-level="2"
        >
          In Your Cart
        </h5>
        <p>
          Your purchase contains the following:
        </p>
        <div
          class="row align-items-center mb-3 product-wrapper"
        >
          <div
            class="col-5"
          >
            <div
              class="embed-responsive embed-responsive-16by9"
            >
              <img
                alt=""
                class="img-thumbnail product-thumbnail"
                src="https://edx.org/eckifega.png"
              />
            </div>
          </div>
          <div
            class="col-7"
          >
            <h6
              aria-level="3"
              class="m-0"
            >
              Introduction to Chickenss
            </h6>
            <p
              class="m-0"
            >
              Verified Certificate
            </p>
          </div>
        </div>
        <form
          class="summary-row form-inline"
        >
          <div
            class="form-group mr-2"
          >
            <label
              for="code-quantity"
            >
              Quantity
            </label>
            <div
              class="position-relative ml-2"
            >
              <div
                class="pgn__form-control-decorator-group"
              >
                <input
                  class="has-value form-control"
                  id="code-quantity"
                  max="100"
                  min="1"
                  name="code-quantity"
                  style="width: 5rem;"
                  type="number"
                  value="1"
                />
              </div>
              <small
                class="text-muted small"
                style="position: absolute; top: 100%; left: 0px;"
              >
                Max: 100
              </small>
            </div>
          </div>
          <button
            aria-disabled="false"
            aria-live="assertive"
            class="pgn__stateful-btn pgn__stateful-btn-state-default btn btn-primary btn-sm"
            type="submit"
          >
            <span
              class="d-flex align-items-center justify-content-center"
            >
              <span>
                Update
              </span>
            </span>
          </button>
        </form>
      </div>
      <div
        aria-labelledby="summary-heading"
        aria-live="polite"
        class="basket-section"
        role="region"
      >
        <h5
          aria-level="2"
          id="summary-heading"
        >
          Summary
        </h5>
         
        <div
          class="summary-row"
        >
          <table
            class="w-100"
          >
            <thead>
              <tr>
                <th
                  class="font-weight-normal"
                >
                  Price
                </th>
                <th
                  class="font-weight-normal"
                >
                  Quantity
                </th>
                <th
                  class="font-weight-normal text-right"
                >
                  Subtotal
                </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td>
                  $92.77
                </td>
                <td>
                  1
                </td>
                <td
                  class="text-right"
                >
                  $92.77
                </td>
              </tr>
            </tbody>
          </table>
        </div>
        <div
          class="summary-row font-weight-bold d-flex"
        >
          <span
            class="flex-grow-1"
          >
            TOTAL
          </span>
          <span
            class="text-right"
          >
            $92.77
          </span>
        </div>
      </div>
      <div
        class="basket-section"
      >
        <h5
          aria-level="2"
        >
          Order Details
        </h5>
        <p>
          By purchasing, you and your organization agree to the following terms:
        </p>
        <ul>
          <li>
            Each code is valid for the one course covered and can be used only one time.
          </li>
          <li>
            You are responsible for distributing codes to your learners in your organization.
          </li>
          <li>
            Each code will expire in one year from date of purchase or, if earlier, once the course is closed.
          </li>
          <li>
            If a course is not designated as self-paced, you should confirm that a course run is available before expiration.
          </li>
          <li>
            You may not resell codes to third parties.
          </li>
          <li>
            All edX for Business sales are final and not eligible for refunds.
          </li>
        </ul>
        <p>
          You will receive an email at person@edx.org with your enrollment code(s).
        </p>
      </div>
    </div>
  </section>
</div>
`;

exports[`<Cart /> renders a cart in non USD currency 1`] = `
<div>
  <section
    aria-label="Shopping Cart Details"
    aria-live="polite"
    aria-relevant="all"
  >
    <div>
      <span
        class="sr-only"
      >
        Shopping cart details are loaded.
      </span>
      <div
        class="basket-section"
      >
        <h5
          aria-level="2"
        >
          In Your Cart
        </h5>
        <p>
          Your purchase contains the following:
        </p>
        <div
          class="row align-items-center mb-3 product-wrapper"
        >
          <div
            class="col-5"
          >
            <div
              class="embed-responsive embed-responsive-16by9"
            >
              <img
                alt=""
                class="img-thumbnail product-thumbnail"
                src="https://edx.org/lu.jpg"
              />
            </div>
          </div>
          <div
            class="col-7"
          >
            <h6
              aria-level="3"
              class="m-0"
            >
              Introduction to Guanacos
            </h6>
            <p
              class="m-0"
            >
              Verified Certificate
            </p>
          </div>
        </div>
      </div>
      <div
        aria-labelledby="summary-heading"
        aria-live="polite"
        class="basket-section"
        role="region"
      >
        <h5
          aria-level="2"
          id="summary-heading"
        >
          Summary
        </h5>
         
        <div
          class="summary-row d-flex"
        >
          <span
            class="flex-grow-1"
          >
            Price
          </span>
          <span
            class="summary-price"
          >
            <span>
              MX$1,264
               *
            </span>
          </span>
        </div>
        <div
          class="summary-row font-weight-bold d-flex"
        >
          <span
            class="flex-grow-1"
          >
            TOTAL
          </span>
          <span
            class="text-right"
          >
            <span>
              MX$1,264
               *
            </span>
          </span>
        </div>
        <div
          class="text-muted font-italic"
        >
          * This total contains an approximate conversion. You will be charged 
          $66.22
           USD.
        </div>
      </div>
      <div
        class="basket-section"
      >
        <h5
          aria-level="2"
        >
          Order Details
        </h5>
        <p>
          After you complete your order you will be able to select course dates from your dashboard.
        </p>
      </div>
    </div>
  </section>
</div>
`;

exports[`<Cart /> renders a cart with an offer 1`] = `
<div>
  <section
    aria-label="Shopping Cart Details"
    aria-live="polite"
    aria-relevant="all"
  >
    <div>
      <span
        class="sr-only"
      >
        Shopping cart details are loaded.
      </span>
      <div
        class="basket-section"
      >
        <h5
          aria-level="2"
        >
          In Your Cart
        </h5>
        <p>
          Your purchase contains the following:
        </p>
        <div
          class="row align-items-center mb-3 product-wrapper"
        >
          <div
            class="col-5"
          >
            <div
              class="embed-responsive embed-responsive-16by9"
            >
              <img
                alt=""
                class="img-thumbnail product-thumbnail"
                src="https://edx.org/lekigud.gif"
              />
            </div>
          </div>
          <div
            class="col-7"
          >
            <h6
              aria-level="3"
              class="m-0"
            >
              Introduction to Pigs
            </h6>
            <p
              class="m-0"
            >
              Verified Certificate
            </p>
          </div>
        </div>
      </div>
      <div
        aria-labelledby="summary-heading"
        aria-live="polite"
        class="basket-section"
        role="region"
      >
        <h5
          aria-level="2"
          id="summary-heading"
        >
          Summary
        </h5>
         
        <div
          class="summary-row d-flex"
        >
          <span
            class="flex-grow-1"
          >
            Price
          </span>
          <span
            class="summary-price"
          >
            $158.87
          </span>
        </div>
        <div
          class="summary-row"
        >
          <p
            class="d-flex m-0"
          >
            <span
              class="flex-grow-1"
            >
              Discounts applied
            </span>
            <span>
              $0.00
            </span>
          </p>
          <p
            class="m-0 text-muted"
          >
            50%
             discount provided by Pied Piper.
          </p>
        </div>
        <div
          class="summary-row font-weight-bold d-flex"
        >
          <span
            class="flex-grow-1"
          >
            TOTAL
          </span>
          <span
            class="text-right"
          >
            $158.87
          </span>
        </div>
      </div>
      <div
        class="basket-section"
      >
        <h5
          aria-level="2"
        >
          Order Details
        </h5>
        <p>
          After you complete your order you will be able to select course dates from your dashboard.
        </p>
      </div>
    </div>
  </section>
</div>
`;

exports[`<Cart /> renders the loading skeleton 1`] = `
<div>
  <section
    aria-label="Shopping Cart Details"
    aria-live="polite"
    aria-relevant="all"
  >
    <div>
      <span
        class="sr-only"
      >
        Loading, please wait...
      </span>
      <div
        class="skeleton py-2 mb-3 w-50"
      />
      <div
        class="skeleton py-2 mb-4 mr-4"
      />
      <div
        class="row align-items-center mb-5"
      >
        <div
          class="col-5"
        >
          <div
            class="skeleton embed-responsive embed-responsive-16by9"
          />
        </div>
        <div
          class="col-7"
        >
          <div
            class="skeleton py-2 mb-3 w-50"
          />
          <div
            class="skeleton py-2 mr-4"
          />
        </div>
      </div>
      <div
        class="skeleton py-2 mb-3 w-50"
      />
      <div
        class="skeleton py-2 mb-2"
      />
      <div
        class="skeleton py-2 mb-5"
      />
      <div
        class="skeleton py-2 mb-3 w-50"
      />
      <div
        class="skeleton py-2 mb-2 mr-4"
      />
      <div
        class="skeleton py-2 mb-5 w-75"
      />
    </div>
  </section>
</div>
`;
